virtual_ship/expedition/verify_schedule.py

```diff
@@ -0,0 +1,130 @@
+"""verify_schedule function and supporting classes."""
+
+from datetime import timedelta
+
+import pyproj
+from parcels import FieldSet
+
+from .input_data import InputData
+from .instrument_type import InstrumentType
+from .schedule import Schedule
+from .ship_config import ShipConfig
+from .waypoint import Waypoint
+
+
+def verify_schedule(
+    projection: pyproj.Geod,
+    ship_config: ShipConfig,
+    schedule: Schedule,
+    input_data: InputData,
+) -> None:
+    """
+    Verify waypoints are ordered by time, first waypoint has a start time, and that schedule is feasible in terms of time if no unexpected events happen.
+
+    :param projection: projection used to sail between waypoints.
+    :param ship_config: The cruise ship_configuration.
+    :param schedule: The schedule to verify.
+    :param input_data: Fieldsets that can be used to check for zero UV condition (is waypoint on land).
+    :raises PlanningError: If waypoints are not feasible or incorrect.
+    :raises ValueError: If there are no fieldsets in the ship_config, which are needed to verify all waypoints are on water.
+    """
+    if len(schedule.waypoints) == 0:
+        raise PlanningError("At least one waypoint must be provided.")
```

Comment:
Are they called waypoints for the users? Else rename here

Reply:
they are called waypoints for the users